src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -16203,18 +16215,21 @@ public static Vector128<double> Invoke(Vector128<double> x)
 
             public static Vector256<double> Invoke(Vector256<double> x)
             {
-                Vector256<ulong> uxMasked = Vector256.Abs(x).AsUInt64();
-                if (Vector256.GreaterThanAny(uxMasked, Vector256.Create(MaxVectorizedValue)))
+                Vector256<double> uxMasked = Vector256.Abs(x);
+                if (Vector256.GreaterThanAny(uxMasked.AsUInt64(), Vector256.Create(MaxVectorizedValue)))
                 {
                     return ApplyScalar<CosOperatorDouble>(x);
                 }
 
-                Vector256<double> r = uxMasked.AsDouble();
                 Vector256<double> almHuge = Vector256.Create(AlmHuge);
-                Vector256<double> dn = (r * Vector256.Create(1 / double.Pi)) + Vector256.Create(double.Pi / 2) + almHuge;
+                Vector256<double> dn = (uxMasked * Vector256.Create(1 / double.Pi)) + Vector256.Create(double.Pi / 2) + almHuge;
                 Vector256<ulong> odd = dn.AsUInt64() << 63;
                 dn = dn - almHuge - Vector256.Create(0.5);
-                Vector256<double> f = r + (dn * Vector256.Create(-double.Pi)) + (dn * Vector256.Create(Pi_Tail2)) + (dn * Vector256.Create(Pi_Tail3));
+                Vector256<double> f =
+                    MultiplyAddEstimateOperator<double>.Invoke(dn, Vector256.Create(-double.Pi),
```

Comment:
Yes. I'm not 100% sure that's desirable, but to my knowledge using MultiplyAddEstimate here is appropriate since it'll give us the more precise answer if accelerated and otherwise just do the non-fused thing. @tannergooding?

Reply:
It should be fine here, yes. There is some potential that it actually performs worse performing depending on how the results compound together, but I think it's fine in this particular case.

If we were using a more exact algorithm, it might be inappropriate because it could change which polynomials we need to use. However, the vectorized algorithm is already non-exact, so this should only reduce the maximum ULP error for most inputs.